arithmetization/src/main/java/net/consensys/linea/zktracer/module/blockdata/Blockdata.java

```diff
@@ -93,7 +94,20 @@ public List<ColumnHeader> columnsHeaders() {
   @Override
   public void commit(List<MappedByteBuffer> buffers) {
     final long firstBlockNumber = operations.getFirst().absoluteBlockNumber();
-    final long chainId = getChainIdFromTransaction(rlpTxn.operations().getLast().tx());
+
+    // TODO: this doesn't work if all transaction of the batch are WO ChainId
+    long chainId = -1;
+    for (RlpTxnOperation tx : rlpTxn.operations().getAll()) {
+      try {
+        chainId = getChainIdFromTransaction(tx.tx());
+      } catch (Exception e) {
+        continue;
+      }
+      if (chainId != -1) {
+        break;
+      }
+    }
+
```

Comment:
Do you need a check after the loop which throws exception if chainId == -1 ?

Reply:
we could, but anyway it would blow in the constraint / when writting the .lt file. But cleaner, I agree, I'll add it.